lib/client.js

```diff
@@ -0,0 +1,97 @@
+const turbo = require('turbo-net')
+const { HTTPParser } = require('http-parser-js')
+const EventEmitter = require('events').EventEmitter
+
+const SEP = ': '
+const EOL = '\r\n'
+
+class Client {
+  constructor (obj) {
+    this._pool = []
+
+    this._host = obj.host || '127.0.0.1'
+    this._port = obj.port || 80
+    this._path = obj.path || '/'
+    this._headers = obj.headers || {}
+    this._method = obj.method || 'GET'
+    this.socket = null
+  }
+
+  _connect (cb) {
+    const self = this
+    const buf = this._alloc()
+    const bus = new EventEmitter()
+    const { _host, _port } = self
+    const socket = turbo.connect(_port, _host)
+
+    self.socket = socket
+
+    sendrequestline()
+    sendhead()
+
+    const parser = new HTTPParser(HTTPParser.RESPONSE)
+
+    parser[HTTPParser.kOnHeadersComplete] = onhead
+    parser[HTTPParser.kOnBody] = onbody
+    parser[HTTPParser.kOnMessageComplete] = onend
+
+    function onhead () {}
+
+    function onbody (body, start, end) {
+      bus.emit('data', body)
```

Comment:
should just call `bus.ondata(body, start, end)` (set bus.ondata to function () {} when you make the emitter)